chapter3/calc/Eval.hs

```diff
@@ -34,6 +34,6 @@ nf :: Expr -> Expr
 nf x = fromMaybe x (nf <$> eval' x)
 
 eval :: Expr -> Maybe Expr
-eval t = case isVal (nf t) of
-  True  -> Just (nf t)
-  False -> Nothing -- term is "stuck"
+eval t = case nf t of
+  nft | isVal nft -> Just nft
+      | otherwise -> Nothing -- term is "stuck"
```

Comment:
Ahh, I see. The version you proposed the line numbers (which are kind of a pain to change) so we'll just go with the guarded version.